### ISSUE TITLE
Strip debug symbols in release

### ISSUE THREAD

**Issue by savchenko** [GothenburgBitFactory/taskchampion#208]
When building via `cargo build --release`, symbols are still present:
```
checksec --file=$HOME/.local/bin/ta
RELRO           STACK CANARY      NX            PIE             RPATH      RUNPATH        Symbols         FORTIFY Fortified       Fortifiable     FILE
Full RELRO      Canary found      NX enabled    PIE enabled     No RPATH   No RUNPATH   19684) Symbols      No    0   12
```
Perhaps good ol' `RUSTFLAGS='-C link-arg=-s' cargo build --release` will do as default release step?

**Comment by djmitche** [comment]
Could this be accomplished in `Cargo.toml`?  https://doc.rust-lang.org/cargo/reference/profiles.html

Aside from binary size, what's the issue with including symbols?

**Comment by savchenko** [comment]
> Aside from binary size, what's the issue with including symbols?

None. But the difference is ~4.2 vs 17.3 Mb.

**Comment by savchenko** [comment]
> Could this be accomplished in Cargo.toml?

"It depends". This should eventually work:

```toml
cargo-features = ["strip"]

[workspace]
members = [
    "taskchampion",
    "cli",
    "sync-server"
]

[profile.release]
strip = "symbols"
```

At present this requires nightly rustc. With `rustc 1.48.0` that is shipped in Debian 11 the feature returns an error:
```
error: the option `Z` is only accepted on the nightly compiler
```

**Comment by savchenko** [comment]
Tried `upx` to see if executable with debug symbols can be efficiently packed and it somewhat surprisingly fails on TC:
```
        File size         Ratio      Format      Name
   --------------------   ------   -----------   -----------
upx: task: CantPackException: bad DT_GNU_HASH n_bucket=0x3a  n_bitmask=0x4  len=0x198
```

**Comment by dbr** [comment]
https://github.com/rust-lang/rust/issues/72110 is the tracking issue for stripping debug info etc from `Cargo.toml`. It being stabilized seems to be largely blocked by macOS support, https://github.com/rust-lang/rust/pull/82037

This seems like a good thing to aim for in a "1.0" release, by which point this feature will hopefully be in Rust stable

**Comment by dbr** [comment]
`rustc -C strip` now stable as of Rust toolchain 1.58, and the strip option is available in cargo as of 1.59,

https://github.com/rust-lang/cargo/blob/master/CHANGELOG.md#cargo-159-2022-02-24
